src/algebra/ordered_monoid.lean

```diff
@@ -903,50 +897,76 @@ lemma min_mul_max [linear_order α] [comm_semigroup α] (n m : α) :
   min n m * max n m = n * m :=
 fn_min_mul_fn_max id n m
 
-section linear_ordered_cancel_add_comm_monoid
-variables [linear_ordered_cancel_add_comm_monoid α]
+/-- A decidable linearly ordered cancellative commutative monoid
+is a commutative monoid with a linear order
+in which multiplication is cancellative and strictly monotone. -/
+@[protect_proj] class linear_ordered_cancel_comm_monoid (α : Type u)
+  extends ordered_cancel_comm_monoid α, linear_order α
+
+/-- A decidable linearly ordered cancellative additive commutative monoid
+is an additive commutative monoid with a decidable linear order
+in which addition is cancellative and strictly monotone. -/
+@[protect_proj] class linear_ordered_cancel_add_comm_monoid (α : Type u)
+  extends ordered_cancel_add_comm_monoid α, linear_order α
+
+attribute [to_additive] linear_ordered_cancel_comm_monoid
+  linear_ordered_cancel_comm_monoid.to_ordered_cancel_comm_monoid
```

Comment:
If you put the `add` version first, you can avoid the separate `attribute` line (unless this interferes with `protect_proj`):

```suggestion
/-- A decidable linearly ordered cancellative additive commutative monoid
is an additive commutative monoid with a decidable linear order
in which addition is cancellative and strictly monotone. -/
@[protect_proj] class linear_ordered_cancel_add_comm_monoid (α : Type u)
  extends ordered_cancel_add_comm_monoid α, linear_order α
  
/-- A decidable linearly ordered cancellative commutative monoid
is a commutative monoid with a linear order
in which multiplication is cancellative and strictly monotone. -/
@[protect_proj, to_additive] class linear_ordered_cancel_comm_monoid (α : Type u)
  extends ordered_cancel_comm_monoid α, linear_order α
```

Reply:
Did you actually try this? It doesn't work for me. I've switched the order anyway.

Reply:
With the classes in the other order I can tag the second class `to_additive` but still have to tag the projection separately (although this is better than what I had).